client/allochealth/tracker.go

```diff
@@ -62,6 +66,10 @@ type Tracker struct {
 	// not needed
 	allocStopped chan struct{}
 
+	// lifecycleTasks is a set of tasks with lifecycle hook set and may
+	// terminate without affecting alloc health
+	lifecycleTasks map[string]bool
```

Comment:
Let's document that sidecars are explicitly *excluded* from this (code suggestions are disabled, sorry)